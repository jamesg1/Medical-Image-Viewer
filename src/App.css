.app {
  text-align: center;
}

.hidden {
  display: none;
}
