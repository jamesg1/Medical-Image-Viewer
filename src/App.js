import React, { Component } from 'react';
import Header from './components/header';
import ImageSeriesViewer from './components/image_series_viewer';
import './App.css';

export const PUBLIC_URL = 'http://localhost:3000';

class App extends Component {

  constructor(props) {
    super(props);
    this.state = {
      loading: true,
      imageResults: [],
      stack: {}
    };
  }

  componentDidMount() {
    fetch('/api/image_series.json', {
      headers : { 
        'Content-Type': 'application/json',
        'Accept': 'application/json'
       }
    })
    .then((response) => response.json())
    .then((response) => {
      this.setState({
        imageResults: response.Results,
        loading: false,
        stack: {
          imageIds: response.Results.map(image => `${PUBLIC_URL}/${image.Dicom}`),
          currentImageIdIndex: 0
        }
      });
    });
  }

  render() {
    const { loading, stack } = this.state;
    return !loading ? (
      <div className="app">
        <Header />
        <ImageSeriesViewer stack={stack} />
      </div>
    ) : null;
  }
}

export default App;
