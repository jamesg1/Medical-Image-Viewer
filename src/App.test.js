import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import App from './App';

describe('App', () => {
  // Default test from Create React App
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders component without props', () => {
    const wrapper = shallow(<App />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
