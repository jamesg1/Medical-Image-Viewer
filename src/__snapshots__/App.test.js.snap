// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders component without props 1`] = `
<div
  className="app"
>
  <Header
    title="Medical Image Series Viewer"
  />
  <h3>
    Loading Please wait ....
  </h3>
</div>
`;
